opentelekomcloud/resource_opentelekomcloud_cce_node_pool_v3.go

```diff
@@ -0,0 +1,526 @@
+				Required: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"size": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"volumetype": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"extend_param": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					}},
+			},
+			"availability_zone": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "random",
+			},
+			"os": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"key_pair": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ExactlyOneOf: []string{"password", "key_pair"},
+			},
+			"password": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				Sensitive:    true,
+				ExactlyOneOf: []string{"password", "key_pair"},
+			},
+			"taints": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"key": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"value": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"effect": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					}},
+			},
+			"billing_mode": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"preinstall": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				StateFunc: func(v interface{}) string {
+					switch v.(type) {
+					case string:
+						return installScriptHashSum(v.(string))
+					default:
+						return ""
+					}
+				},
+			},
+			"postinstall": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				StateFunc: func(v interface{}) string {
+					switch v.(type) {
+					case string:
+						return installScriptHashSum(v.(string))
+					default:
+						return ""
+					}
+				},
+			},
+			"subnet_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"scale_enable": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"min_node_count": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"max_node_count": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"scale_down_cooldown_time": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"priority": {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceCCENodePoolCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	nodePoolClient, err := config.cceV3Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("Error creating Ope Telekom Cloud CCE Node Pool client: %s", err)
```

Comment:
Here and further.
It's considered a better code style to not start error string with a capital letter:
https://github.com/golang/go/wiki/CodeReviewComments#error-strings